tests/Transport/AwsAuthV4Test.php

```diff
@@ -205,4 +205,35 @@ public function testSignsWithEnvironmentalCredentials(): void
             );
         }
     }
+
+    /**
+     * @group unit
+     * @depends testSignsWithProvidedCredentials
+     */
+    public function testStripsTrailingDotInHost(): void
+    {
+        $host = $this->_getHost();
+        $hostWithTrailingDot = $host.'.';
+
+        $config = [
+            'persistent' => false,
+            'transport' => 'AwsAuthV4',
+            'aws_access_key_id' => 'foo',
+            'aws_secret_access_key' => 'bar',
+            'aws_session_token' => 'baz',
+            'aws_region' => 'us-east-1',
+            'host' => $hostWithTrailingDot,
+        ];
+        $client = $this->_getClient($config);
+
+        try {
+            $client->request('_stats');
```

Comment:
Is this executing all the new code? Maybe that is the issue?

Reply:
We're able to have a look later on today or tomorrow.

Reply:
So, we hacked the make targets to be able to generate phpunit code coverage reports, and it doesn't look too bad there.. see attached screenshot. 🤔

![Screenshot 2022-08-25 at 12-23-31 Code Coverage for var www html src Transport AwsAuthV4 php](https://user-images.githubusercontent.com/11797354/186642717-8ee18429-ae29-4061-9c9d-76c72775ecec.png)